buildSrc/src/main/kotlin/com/jakewharton/rxbinding/project/KotlinGenTask.kt

```diff
@@ -171,162 +219,151 @@ open class KotlinGenTask : SourceTask() {
    * Represents a kotlin file that corresponds to a Java file/class in an RxBinding module
    */
   class KFile {
-    var fileName: String by Delegates.notNull<String>()
-    var packageName: String by Delegates.notNull<String>()
-    var bindingClass: String by Delegates.notNull<String>()
-    var extendedClass: String by Delegates.notNull<String>()
+    var fileName: String by Delegates.notNull()
+    var packageName: String by Delegates.notNull()
+    var bindingClass: String by Delegates.notNull()
+    var extendedClass: String by Delegates.notNull()
     val methods = mutableListOf<KMethod>()
-    val imports = mutableListOf<String>()
 
     /** Generates the code and writes it to the desired directory */
     fun generate(directory: File) {
-      var directoryPath = directory.absolutePath
-      var finalDir: File? = null
-      if (!packageName.isEmpty()) {
-        packageName.split('.').forEach {
-          directoryPath += File.separator + it
-        }
-        finalDir = File(directoryPath)
-        Files.createDirectories(finalDir.toPath())
-      }
-
-      File(finalDir, fileName).bufferedWriter().use { writer ->
-        writer.append("package $packageName\n\n")
-
-        imports.forEach { im ->
-          if (!IGNORED_IMPORTS.contains(im)) {
-            writer.append("import $im\n")
+      KotlinFile.builder(packageName, fileName)
+          .apply {
+            methods.firstOrNull { it.emitsUnit() }?.let {
+              addStaticImport("com.jakewharton.rxbinding2.internal", "VoidToUnit")
+            }
+            methods.map { it.generate(ClassName.bestGuess(bindingClass)) }
+                .forEach { addFun(it) }
           }
-        }
-        methods.firstOrNull { it.emitsUnit() }?.let {
-          writer.append("import com.jakewharton.rxbinding2.internal.VoidToUnit\n")
-        }
+          // @file:Suppress("NOTHING_TO_INLINE")
+          .addFileAnnotation(AnnotationSpec.builder(Suppress::class)
+              .useSiteTarget(FILE)
+              .addMember("names", "%S", "NOTHING_TO_INLINE")
+              .build())
+          .skipJavaLangImports(true)
```

Comment:
Remove? Does this actually do anything?

Reply:
This feature did that in Java, but you shouldn't use it, and we're not going to ship one for Kotlin!